satpy/tests/reader_tests/test_avhrr_l1b_gaclac.py

```diff
@@ -506,3 +466,107 @@ def test__slice(self, strip_invalid_lat, get_qual_flags):
         pygac.utils.slice_channel.return_value = ('sliced', 'foo', 'bar')
         data_slc = fh._slice(data)
         self.assertEqual(data_slc, 'sliced')
+
+
+class TestGetDataset(ModulePatcher):
+    """Test the get_dataset method."""
+
+    def setUp(self):
+        """Set up the instance."""
+        self.exp = xr.DataArray(da.ones((3, 3)),
+                                name='1',
+                                dims=('y', 'x'),
+                                coords={'acq_time': ('y', [0, 1, 2])},
+                                attrs={'name': '1',
+                                       'platform_name': 'spacecraft_name',
+                                       'orbit_number': 123,
+                                       'sensor': 'sensor',
+                                       'foo': 'bar',
+                                       'standard_name': 'my_standard_name'})
+        self.exp.coords['acq_time'].attrs['long_name'] = 'Mean scanline acquisition time'
+        super().setUp()
+
+    @mock.patch('satpy.readers.avhrr_l1b_gaclac.GACLACFile.__init__', return_value=None)
+    @mock.patch('satpy.readers.avhrr_l1b_gaclac.GACLACFile.read_raw_data')
+    @mock.patch('satpy.readers.avhrr_l1b_gaclac.GACLACFile._get_channel', return_value=np.ones((3, 3)))
+    def test_get_dataset_channels(self, get_channel, *mocks):
+        """Test getting the channel datasets."""
+        pygac_reader = _get_reader_mocked()
+        fh = self._create_reader(pygac_reader)
+
+        # Test calibration to reflectance as well as attributes.
+        key, res = self._get_dataset(fh)
+        exp = self._create_expected(res.name)
+        exp.attrs['orbital_parameters'] = {'tle': 'tle'}
+
+        xr.testing.assert_identical(res, exp)
+        get_channel.assert_called_with(key)
+
+        self._check_get_channel_calls(fh, get_channel)
+
+    @staticmethod
+    def _get_dataset(fh):
+        from satpy.tests.utils import make_dataid
+
+        key = make_dataid(name='1', calibration='reflectance')
+        info = {'name': '1', 'standard_name': 'my_standard_name'}
+        res = fh.get_dataset(key=key, info=info)
+        return key, res
+
+    @staticmethod
+    def _create_reader(reader):
```

Comment:
Much better now, thanks for refactoring this! Just one tiny nitpick:

```suggestion
    def _create_file_handler(reader):
```

Reply:
Oh and if you are very motivated you could move the other `test_get_dataset` methods to the `TestGetDataset` class. But I don't want to ask too much refactoring of my test code :)  So really only if you have time.

Reply:
Well, time is really a scarce resource atm, so I'll leave this as it is for now...